app/controllers/main_controller.rb

```diff
@@ -8,4 +8,30 @@
 
 # Your controller actions go below this line.
 # -----------------------------------------------------------------------------
+# require_relative "../lib/dictionary"
+# require_relative "../lib/string_segmenter"
 
+MyApp.get "/" do
+  erb :"homepage"
+end
+
+MyApp.get "/admin" do
+  erb :"/a/admin"
+end
+
+
+
+
+MyApp.get "/segmenter" do
+  
+  @segment = StringSegmenter.new(params[:string])
+
+  @i = Letter.new
+  @i.search_string = params[:string]
+  @i.segmented_parts = segmented_words_as_strings
```

Comment:
`segmented_words_as_strings`... This was in my answer, yes. But I had it as a reference pointer to something that had already been defined.

You're not far off track. Let me know if the other comment I've made and this one aren't helpful.